src/mscorlib/src/System/String.cs

```diff
@@ -973,17 +973,20 @@ internal String[] SplitInternal(char[] separator, int count, StringSplitOptions
 
             if ((count == 0) || (omitEmptyEntries && this.Length == 0)) 
             {           
-                return new String[0];
+                return EmptyArray<String>.Value;
+            }
+
+            if (count == 1)
+            {
+                return new String[] { this };
```

Comment:
This count == 1 optimization seems like a fine change to make, regardless of what's decided about the count == 0 case.  So even if we decide not to do the count == 0 optimization, this one seems worthwhile and without compat risk.